src/controllers/auth.js

```diff
@@ -0,0 +1,162 @@
+const bcrypt = require('bcrypt');
+const { Volunteer, Organization } = require('../models/user');
+
+exports.generalsignin = async (req, res) => {
+  const { email, password } = req.body;
+
+  const volunteer = await Volunteer.findOne({ email });
+  const organization = await Organization.findOne({ email });
+
+  const volunteerpassword = await Volunteer.findOne({ password });
+  const organizationpassword = await Organization.findOne({ password });
+
+  let volunteerororganization = null;
+
+  if (volunteer && volunteerpassword) {
+    volunteerororganization = volunteer;
+  } else if (organization && organizationpassword) {
+    volunteerororganization = organization;
+  }
+
+  if (!volunteerororganization) {
+    return res.status(400).json({ error: 'Wrong username,email or password' });
+  }
```

Comment:
If you do the change I mentioned at the top this can be replace with:
```suggestion
  if (!user) {
    return res.status(400).json({ error: 'Could not find user' });
  }
```